lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -66,6 +68,23 @@ def object? object
       object.type.include? RDFVocabularies::PCDMTerms.Object
     end
 
+    # TODO: This is ugly. Are there clever ways to assemble a list of ancestors for a 
+    # given collection?
+    def collection_ancestor? collections
+      next_batch = []
+      current_collections = collections
+      while !current_collections.empty? do
+        current_collections.each do |check|
+          check.collections.each do |c|
+            return true if ActiveFedora::Base.find(c.id).collections.map { |x| x.id == self.id }
```

Comment:
Isn't this going to make it really slow to add collections to other collections?  Do we care?  Does it make sense to optimize this to query the index directly rather than loading all ancestors of all the collections involved?